pkg/server/image.go

```diff
@@ -338,61 +338,90 @@ var singleResize singleflight.Group
 
 func (ih *ImageHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request, file blob.Ref) {
 	ctx := req.Context()
-	if !httputil.IsGet(req) {
-		http.Error(rw, "Invalid method", http.StatusBadRequest)
+	res, err := ih.Search.Describe(ctx, &search.DescribeRequest{
+		BlobRef: file,
+		Depth:   1,
+	})
+	if err != nil {
+		http.Error(rw, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	mw, mh := ih.MaxWidth, ih.MaxHeight
-	if mw == 0 || mh == 0 || mw > search.MaxImageSize || mh > search.MaxImageSize {
-		http.Error(rw, "bogus dimensions", http.StatusBadRequest)
+
+	db, ok := res.Meta[file.String()]
+	if !ok {
+		http.Error(rw, "blob meta not found", http.StatusInternalServerError)
 		return
 	}
 
-	key := cacheKey(file.String(), mw, mh)
-	etag := blob.RefFromString(key).String()[5:]
-	inm := req.Header.Get("If-None-Match")
-	if inm != "" {
-		if strings.Trim(inm, `"`) == etag {
-			thumbCacheHeader304.Add(1)
-			rw.WriteHeader(http.StatusNotModified)
+	var imageData []byte
+	var etag string
+	if db.File.MIMEType == "image/svg+xml" {
+		_, etag, _ = strings.Cut(file.String(), "-")
+		fr, err := schema.NewFileReader(ctx, ih.Fetcher, file)
+		if err != nil {
+			http.Error(rw, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		defer fr.Close()
+		imageData, err = io.ReadAll(fr)
+		if err != nil {
+			http.Error(rw, err.Error(), http.StatusInternalServerError)
 			return
 		}
 	} else {
-		if !disableThumbCache && req.Header.Get("If-Modified-Since") != "" {
-			thumbCacheHeader304.Add(1)
-			rw.WriteHeader(http.StatusNotModified)
+		if !httputil.IsGet(req) {
+			http.Error(rw, "Invalid method", http.StatusBadRequest)
+			return
+		}
```

Comment:
shouldn't this be up earlier, before the mime type check? doesn't seem svg-vs-not specific?